frontend/src/context/RequireAuth.tsx

```diff
@@ -0,0 +1,20 @@
+import { Navigate, useLocation } from "react-router-dom";
+import { useUser } from "./UserContext";
+
+function RequireAuth({ children }: { children: JSX.Element }) {
```

Comment:
I don't think this needs to be a component, it could be a hook.

I think it will interfere with loading states (if we add loading states later)